### ISSUE TITLE
Taskforce 0.2.0 release

### ISSUE THREAD

**Issue by michael** [michael/hub#28]
We should take the change and take some time to organise better our codebase for the 0.2.0 release, before we look into implementing new features.

Here's an attempt of most important features I see for the 0.2.0 release

Composer
- Improved build system (@oliver----)
- Plugin System (@michael, @vectorsize)
- New Content Types (images, links)
- Support Networks on the Composer side

Hub
- Code cleanup, can we dump legacy code (should we do the switch to EJS)
- How can we have a better data layer (still against using an ORM as writing SQL directly has a lot of advantages imho)
- DB Migration system (in house, declarative but not to far away from sql statements)
- Focus on the presentation layer (optimize how documents look like and make reading a doc on Substance.io a real pleasure)
- Networks (yeah this plays into the usability discussion)

Will update this issue as thoughts come up. Let's have the discussion below. In general.. let's polish before looking into new features? Do you guys agree?

**Comment by vectorsize** [comment]
Totally agree with you @michael. we should polish before looking into new features.
So after todays talk and keeping what's relevant from michael's previous post:

## General discussion
- Talker app is the new sh*t, now we have an internals discussion room too (The reason I brought Talker app back to the discussion was that constant availability on Skype was really breaking my productivity, and I read this post [ HOW GITHUB WORKS: BE ASYNCHRONOUS](http://zachholman.com/posts/how-github-works-asynchronous/) not so long ago and it all made sense)
  - We can leave mentioned comments for people missing in the conversation and they can always come back and read it when they have time 
  - Sending messages through /msg works only when users are online, and they disappear after logging out.
- Plugin functionality gets postponed. As long as the api works for us to add new nodes it's not a priory
- @yuchi said that the user shouldn't have to go to the hub, and that makes total sense, however, as I said some times before, if at some point Substance would have to charge users, say..for hub usage, then the more useful stuff is on the server the easier it will be for people to understand they have to pay, so we have to think strategically there too. I just mention this so that we have it in mind.

## Composer
- Improved build system
- Support Networks on the Composer side
- Link node type
  - We still need to decide how the UI will be for the url metadata
  - Also keep in mind that one might want to link to headings in the current doc
- Image node type
  - Image node type should have a caption
  - Once caption is solved interface-wise do we want other data for the image? like title?
  - how much of a priority would having annotations on the image be? I'm sure it's not very hard to implement a flickr-like system

## Hub
- Regarding le hub we agreed on having the basic functionality in place for ourselves to do simple hub maitainance stuff.
- Code cleanup, can we dump legacy code (should we do the switch to EJS) // +1 to this from me
- How can we have a better data layer (still against using an ORM as writing SQL directly has a lot of advantages imho) // +1 to a better data layer
- DB Migration system (in house, declarative but not to far away from sql statements) // +1 to migrations, YES!
- Focus on the presentation layer (optimize how documents look like and make reading a doc on Substance.io a real pleasure)
- Networks (yeah this plays into the usability discussion)
- Once the hub starts being used, besides this managing tools, we also should look into the terms of usage as discussed [here](https://github.com/michael/hub/issues/25). As @yuchi said, it's lawyers territory, but while we don't have any lawyer with us nothing stops us from using a Creative Commons licensed standard terms like the ones @michael pointed which are taken from [wordpress terms](https://en.wordpress.com/tos/). In cases like people starts using the hub as a pastebin of ilegal stuff and such, we need to cover our asses.

## Substance Camp

we decided on a couple of options regarding dates for the camp. I can't remember them now :/ but it all loks like it will be either after April.

## My Conclusion

All that said, I can help on the composer side and hub, I think @yuchi you know what's to do on the server, so if you see where I could help just say it, I won't start doing stuff by myself at this point. Regarding the composer, @michael lets coordinate the new node types and let me know if you want help with networks and such.

and @oliver---- really sad that we didn't see you today... I hope you get better mmmhhkaayy?? :)

**Comment by yuchi** [comment]
Great sum @vectorsize! ASAP I'm going to create specific issues for every task. We can then prioritize them and start to tackle them with the triage (add tag, assign, put in milestone).

**Comment by michael** [comment]
Nice summary!

I'm getting ready basic image support today. It's looking good. I'll just implement the most critical parts today. We can add support for captions later.

Also thought about link support (UI-wise). Will draw some sketches in the coming days, implementation will be over to you, @vectorsize. 

Regarding plugins: After talking yesterday I realized we should first get all functionality in, then restructure the code and extract a plugin-system.

@yuchi Will also try to think through what we need for the admin panel, I have some idaes which I'm going to sketch as soon as I have some time. Then there's networks, what do you think. Should we start with adding networks or is the admin panel more important?

Roadmap:

I'm for smaller increments. I'd like to push out image and link support as soon as possible. And call this the composer 0.2.0 release.

0.2.0 (image and link support for composer and hub in 4-6 weeks)
0.3.0 (network support in 8-10weeks)

But before we start...  let's take care of the 0.1.0 backlog. https://github.com/michael/hub/issues/27

**Comment by michael** [comment]
Oh there's another priority for 0.2.0 "User Profiles". I guess it's important that people feel at home on Substance, so user profiles need to be pimped (avatar, and some more details regarding the person, followed by all publications).

**Comment by yuchi** [comment]
To be tracked and triaged ( **Hub only** )
- [x] Switch to EJS
- [x] Minimum viable admin panel "framework"
- [ ] Dump/comment-out legacy code (eg: signup/login/preferences)
- [ ] User profiles
